vm/src/obj/objbyteinner.rs

```diff
@@ -426,6 +426,69 @@ impl PyByteInner {
         }
     }
 
+    pub fn setitem(
+        &mut self,
+        needle: Either<PyIntRef, PySliceRef>,
+        object: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        match needle {
+            Either::A(int) => {
+                if let Some(idx) = self.elements.get_pos(int.as_bigint().to_i32().unwrap()) {
+                    let result = match_class!(object,
+                    i @ PyInt => {
+                        if let Some(value) = i.as_bigint().to_u8() {
+                            Ok(value)
+                        }else{
+                            Err(vm.new_value_error("byte must be in range(0, 256)".to_string()))
+                        }
+                    },
+                    _ => {Err(vm.new_type_error("an integer is required".to_string()))}
+                    );
+                    match result {
+                        Ok(value) => {
+                            self.elements[idx] = value;
+                            Ok(vm.new_int(value))
+                        }
+                        Err(error) => Err(error),
```

Comment:
This is a case of error propagation. Please use the `?` operator in this case to collapse the `match` statement into the simpler form:

```rust
let value = result?;
self.elements[idx] = value;
Ok(vm.new_int(value))
```

Reply:
Thanks, I improved the code as suggested.